### ISSUE TITLE
Add field names object

### ISSUE THREAD

**Issue by kalmanb** [movio/movio-apidoc-generator#20]
_From @jamesk on March 9, 2016 23:3_

It would be nice to be able to get the `JSON` representation of a case class's field. 

Example use case:
Say we have a spec which gives us the following case class:

```
case class MyModel(field1: String, field2: String)
```

We want to store this in elasticsearch and later query it. To do so we would use the generated serialiser to get a `JSON` representation which we save to elasticsearch. If we then want to query that data from our code, say for all records where `field2 == "hello"` then we need to use a magic string `"field2"` to build that elasticsearch query. If our spec changes (or maybe we use a different/updated generator) such that `field2` is renamed to `property2` we would introduce a run-time error because of our magic string referring to `"field2"` instead of `"property2`".

It would be better if the field names in the (de)serialiser were available as values, e.g.

```
object MyModel {
  object Fields {
    field1 = "field1"
    field2 = "field2"
  }
}

implicit def jsonWritesMyServiceMyModel[T: Writes]: play.api.libs.json.Writes[MyModel] = {
      (
        (__ \ MyModel.Fields.field1).write[String] and
        (__ \ MyModel.Fields.field2).write[String]
      )(unlift(MyModel.unapply _))
    }
```

Then when building my elasticsearch query we can directly refer to `MyModel.Fields.field1` rather than a magic string. Now we would get a compile time error in our query code.

_Copied from original issue: movio/apidoc-generator-old#21_

**Comment by kalmanb** [comment]
_From @simon-movio on March 10, 2016 18:32_

@jamesk That sounds like a good approach to me :) :+1:

**Comment by kalmanb** [comment]
Fixed in #22
